src/runtime_src/core/tools/common/ReportHost.cpp

```diff
@@ -58,6 +59,16 @@ ReportHost::getPropertyTree20202( const xrt_core::device * /*_pDevice*/,
   _pt.add_child("host", pt);
 }
 
+void
+updateDataMapValue(std::map<std::string, size_t>& data_length_map,
+                  const boost::property_tree::ptree& dev,
+                  const std::string& key)
+{
+  if (data_length_map.find(key) == data_length_map.end())
+    data_length_map[key] = dev.get<std::string>(key).size();

```

Comment:
Use iterator to avoid double hashing

Reply:
I think that might be excessive considering only 4 elements are placed inside the map. I used a map instead of a vector so it is easy to link the key string with the maximum data length. Using an iterator as the key, which would prevent the double hashing, would make everything a little harder to read. I am happy to implement it but it does defeat the purpose of why I used the map.

Reply:
In worst case you are searching 3 times for same key.  I meant maybe you want to use 
```
auto itr = data_length_map.find(key);
if (itr == data_length_map.end())
   data_length_map[key] = ...;
else 
   (*itr).second = std::max((*itr).second, ...);
```
or if your use case implies that 0 is uninitialized then just
```
auto& value = data_length_map[key];
value = value ? ... : ...;

```
But as you say, for your use-case  it's not really important to optimize this; just looks strange.